src/Maker/Security/AbstractSecurityMaker.php

```diff
@@ -0,0 +1,83 @@
+use Symfony\Bundle\MakerBundle\FileManager;
+use Symfony\Bundle\MakerBundle\Maker\AbstractMaker;
+use Symfony\Bundle\MakerBundle\Security\InteractiveSecurityHelper;
+use Symfony\Bundle\MakerBundle\Security\SecurityConfigUpdater;
+use Symfony\Bundle\MakerBundle\Security\SecurityControllerBuilder;
+use Symfony\Bundle\MakerBundle\Util\YamlSourceManipulator;
+use Symfony\Bundle\MakerBundle\Validator;
+use Symfony\Bundle\SecurityBundle\SecurityBundle;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+abstract class AbstractSecurityMaker extends AbstractMaker
+{
+    protected const SECURITY_CONFIG_PATH = 'config/packages/security.yaml';
+
+    protected YamlSourceManipulator $ysm;
+    protected string $securityControllerName;
+    protected string $firewallToUpdate;
+    protected string $userClass;
+    protected string $userNameField;
+    protected bool $willLogout;
+
+    public function __construct(
+        protected FileManager $fileManager,
+        protected SecurityConfigUpdater $securityConfigUpdater,
+        protected SecurityControllerBuilder $securityControllerBuilder,
+    ) {
+    }
+
+    public function configureDependencies(DependencyBuilder $dependencies): void
+    {
+        $dependencies->addClassDependency(SecurityBundle::class, 'security');
+        $dependencies->addClassDependency(Process::class, 'process');
+        $dependencies->addClassDependency(Yaml::class, 'yaml');
+        $dependencies->addClassDependency(DoctrineBundle::class, 'orm');
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        if (!$this->fileManager->fileExists(self::SECURITY_CONFIG_PATH)) {
+            throw new RuntimeCommandException(sprintf('The file "%s" does not exist. PHP & XML configuration formats are currently not supported.', self::SECURITY_CONFIG_PATH));
+        }
+
+        $this->securityControllerName = $io->ask(
+            'Choose a name for the controller class (e.g. <fg=yellow>ApiLoginController</>)',
+            'ApiLoginController',
+            [Validator::class, 'validateClassName']
+        );
+
+        $this->ysm = new YamlSourceManipulator($this->fileManager->getFileContents(self::SECURITY_CONFIG_PATH));
+        $securityData = $this->ysm->getData();
+
+        $securityHelper = new InteractiveSecurityHelper();
+        $this->firewallToUpdate = $securityHelper->guessFirewallName($io, $securityData);
+        $this->userClass = $securityHelper->guessUserClass($io, $securityData['security']['providers']);
+        $this->userNameField = $securityHelper->guessUserNameField($io, $this->userClass, $securityData['security']['providers']);
+        $this->willLogout = $io->confirm('Do you want to generate a \'/logout\' URL?');
+    }
+}
```

Comment:
We could probably find more repeated code between `make:json` & `make:form` logins if we looked hard enough...